repotracker/github_poller.go

```diff
@@ -213,12 +213,24 @@ func (gRepoPoller *GithubRepositoryPoller) GetRevisionsSince(revision string, ma
 				revision, baseRevision)
 		}
 
-		gRepoPoller.ProjectRef.RepotrackerError = revisionDetails
-		if err = gRepoPoller.ProjectRef.Upsert(); err != nil {
-			return []model.Revision{}, errors.Wrap(err, "unable to update projectRef revision details")
+		if setRevision && baseRevision != "" {
```

Comment:
This might be puzzling for a later reader. Could you add a comment explaining the differing behavior, and mentioning that the UI and API have different behavior?